mm2src/coins/nft/nft_structs.rs

```diff
@@ -817,15 +817,28 @@ pub struct ClearNftDbReq {
 #[derive(Clone, Debug, Serialize)]
 pub struct NftInfo {
     /// The address of the NFT token.
-    pub(crate) token_address: Address,
+    pub token_address: Address,
     /// The ID of the NFT token.
     #[serde(serialize_with = "serialize_token_id")]
+    pub token_id: BigUint,
+    /// The blockchain where the NFT exists.
+    pub chain: Chain,
+    /// The type of smart contract that governs this NFT.
+    pub contract_type: ContractType,
+    /// The quantity of this type of NFT owned. Particularly relevant for ERC-1155 tokens,
+    /// where a single token ID can represent multiple assets.
+    pub amount: BigDecimal,
+}
+
+/// Represents the Swap information about a Non-Fungible Token (NFT).
+#[derive(Clone, Debug, Serialize)]
+pub struct NftSwapInfo {
+    /// The address of the NFT token.
+    pub(crate) token_address: Address,
+    /// The ID of the NFT token.
     pub(crate) token_id: BigUint,
     /// The blockchain where the NFT exists.
     pub(crate) chain: Chain,
     /// The type of smart contract that governs this NFT.
     pub(crate) contract_type: ContractType,
```

Comment:
another duplications here.. can you make one extend another if the fields won't change in the future?

Reply:
removed this version of `NftSwapInfo` from `nft_structs.rs`, strange that clippy didnt warn that it wasnt used